src/conventions/h-group/clue-interpretation/connecting-cards.js

```diff
@@ -159,15 +160,75 @@ function find_unknown_connecting(game, giver, target, reacting, identity, connec
 		}
 		// Finessed card is delayed playable
 		else if (game.level >= LEVEL.INTERMEDIATE_FINESSES && state.play_stacks[finesse.suitIndex] + 1 === finesse.rank) {
-			// Could be duplicated in giver's hand - disallow hidden finesse
-			if (state.hands[giver].some(c => c.clued && game.players[giver].thoughts[c.order].inferred.has(identity))) {
+			let bluff = false;
+			if (game.level >= LEVEL.BLUFFS) {
+				// This may be a bluff. resolve_bluff will rule out a bluff if it doesn't
+				// look like it could lead to the following plays.
+				bluff =
+					connected.length == 1 &&
+					((giver + 1) % state.numPlayers) == reacting &&
+					!common.thinksLoaded(state, reacting);
```

Comment:
Simply checking if the reacting player is loaded seems too broad. I think the reacting player only cannot have finesses that have room to be layered on them, but I'm not too sure. (I think if they have a card that could be prompted instead they wouldn't even get to this point.)

Reply:
Agreed, I think `unknown_plays` is the right variable? I've added this and a test that it can bluff with known plays.

Reply:
I think this is closer, but not quite there yet. `unknown_plays` contains the cards that the player knows are delayed playable without knowing their identity (i.e. the card has more than 1 inference). However, a player can still be bluffed even with unknown plays-- I think the only issue is that the reacting player won't bluff if a finesse on them could be layered. See these tests which currently fail:
```js
it('understands a bluff on top of unknown plays that cannot match', () => {
	const game = setup(HGroup, [
		['xx', 'xx', 'xx', 'xx'],
		['r1', 'p3', 'y1', 'y1'],
		['g5', 'b2', 'r3', 'y5'],
		['b4', 'p2', 'g3', 'r5'],
	], {
		level: 11,
		play_stacks: [0, 0, 1, 0, 2],
		starting: PLAYER.DONALD
	});
	takeTurn(game, 'Donald clues 1 to Bob');
	// Since Bob is only queued on 1s, Alice should be able to bluff Bob's p3 using g3.
	takeTurn(game, 'Alice clues 3 to Donald');

	assert.equal(game.common.thoughts[game.state.hands[PLAYER.BOB][1].order].finessed, true);
});

it(`doesn't bluff on top of unknown queued cards`, () => {
	const game = setup(HGroup, [
		['xx', 'xx', 'xx', 'xx', 'xx'],
		['r2', 'g1', 'y1', 'y1', 'r4'],
		['p2', 'b2', 'r3', 'y5', 'y4'],
		['g1', 'g2', 'g3', 'g5', 'p4'],
	], {
		level: 11,
		play_stacks: [1, 0, 0, 0, 0],
		starting: PLAYER.DONALD
	});
	takeTurn(game, 'Donald clues red to Cathy');
	// With r2 queued, we cannot bluff the g1.
	const { play_clues } = find_clues(game);
	const bluff_clues = play_clues[2].filter(clue => {
		return clue.type == CLUE.RANK && clue.target == 2 && clue.value == 2 ||
			clue.type == CLUE.COLOUR && clue.target == 2 && (clue.value == 3 || clue.value == 4);
	});
	assert.equal(bluff_clues.length, 0);
});
```

Maybe something like `!state.hands[reacting].some((c, index) => common.thoughts[c.order].finessed && state.hands[reacting].some((c2, index2) => index2 > index && !c2.clued));`? Which passes both tests above, but doesn't take negative information on cards into consideration. It's kind of a weird situation, for example, if you're playing into a y2 layered finesse (i.e. not from slot 1), and the only other unclued card is a chop moved card that is either 4 or 5, and you are called to play g1. Then you know that you can just play slot 1, but this expression doesn't account for that.

But it could be even more awkward if the "y2" card could be g1 and the other unclued card could be y2, because then the layer is a possibility again, and the bluff can't be given. I think it's something like if the union of both cards have both identities that are being looked for, then a bluff can't be given, but I'm worried that the finessed card may not even have a single identity (e.g. if they are in a superposition of finesses due to an unknown self-component, but maybe this situation isn't possible). It's difficult to think about, not sure if you have any ideas or if you think we should just fail to interpret these situations. :sweat_smile:

Reply:
I added these tests and they pass with my latest patch. The condition now is that they think their unknown card matches the requested identity.

Reply:
Hmm, that still doesn't sound quite right to me. Players don't "know" their own cards other than from a rewind, so this would pretty much always be false. In fact, removing this expression completely still passes all the current tests.

Currently, it seems that any cards that might be played instead will be added to `connected` first, making the length greater than 1 and failing the earlier condition anyways. I think it makes sense since playable/prompt connections are always prioritized over finesse connections, and if the prompt isn't playable then it will trigger the wrong prompt line. But I'm not sure, was this line necessary before?